File: leetgo/arrays/is_anagram.go
package arrays 

func isAnagram(source, target string) bool{
	if len(source) != len(target){
		return false 
	}

	var freq [26]int 

	for idx := 0; idx < len(source); idx++{
		freq[source[idx] - 'a']++ 
		freq[target[idx] - 'a']--
	}

	for idx := 0; idx < len(freq); idx++{
		if freq[idx] != 0{
			return false 
		}
	}
	return true 
}

File: leetpy/arrays_and_hashing/is_anagram.py
def isAnagram(source : str, target : str) -> bool: 
    if len(source) != len(target): 
        return False 
    
    count_source, count_target = {}, {} 
    for i in range(len(source)): 
        count_source[source[i]] = 1 + count_source.get(source[i], 0) 
        count_target[target[i]] = 1 + count_target.get(target[i], 0) 
    return count_source == count_target 